### PR TITLE
Fix invalid signature on hardhat run & Unsupported chain ID: undefined on starknet-invoke/call

### LINKED ISSUE #131: Invalid signature error for account invoke
### Summary
When trying to execute an `account.invoke` outside of the test scripts (Ex: `node file.js`) I am getting the following invalid signature error. When running the same code inside of a test (`npx hardhat test`) it works no problem.

```
Invoke transaction 0x4879f2e799c3f582947033d71f7aaaef212df254b92671cbe783ee0c229411f is REJECTED.
Transaction rejected. Error message:

/home/badurina/projects/cairo-contracts/env/lib/python3.8/site-packages/starkware/cairo/common/signature.cairo:12:5: Error at pc=0:134:
Signature (938829697521780319332146409030067216646503173912283525226354363451947159347, 539935339141187093863590712174480503359402253748499857344377649991277935741), is invalid, with respect to the public key 2749374374727211637233520590806508110892759908750735217418460050042256624526, and the message hash 2048874378711897586683480876739342048796352607406708712693589312404693008671.
Cairo traceback (most recent call last):
src/openzeppelin/account/Account.cairo:94:6: (pc=0:719)
src/openzeppelin/account/Account.cairo:106:36: (pc=0:663)
/home/badurina/projects/cairo-contracts/src/openzeppelin/account/library.cairo:179:26: (pc=0:368)
/home/badurina/projects/cairo-contracts/src/openzeppelin/account/library.cairo:135:9: (pc=0:321)
```

### Environment Details
- dockerized version of devnet `shardlabs/starknet-devnet:0.2.4-arm`
- runtime(tried both): `venv: 'active'` and `dockerizedVersion: 'shardlabs/cairo-cli:0.9.0-arm'`

### Hardhat config
```
require('@shardlabs/starknet-hardhat-plugin');


module.exports = {
  mocha: {
    bail: true,
    timeout: 600_000
  },
  networks: {
    dockerStark: {
      url: 'http://localhost:9000'
    }
  },
  paths: {
    cairoPaths: [ './lib' ]
  },
  starknet: {
    network: 'dockerStark',
    // venv: 'active',
    dockerizedVersion: 'shardlabs/cairo-cli:0.9.0-arm',
    wallets: {
      MyWallet: {
        accountName: 'OpenZeppelin',
        modulePath: 'starkware.starknet.wallets.open_zeppelin.OpenZeppelinAccount',
        accountPath: '~/.starknet_accounts'
      }
    }
  }
};
```

### Script code
```
const { starknet } = require('hardhat');
const { stark: { randomAddress } } = require('starknet');

const done = function (error) {
  if (error) console.error(error);
  console.info('done');
  process.exit();
}

const main = async function () {
  const account = await starknet.deployAccount('OpenZeppelin');
  const contractFactory = await starknet.getContractFactory('Proxy');
  const contract = await contractFactory.deploy({ proxy_admin: account.address });
  const class_hash = randomAddress();
  await account.invoke(contract, 'setClassHash', { class_hash });
};


main()
.then(() => done())
.catch(done);
```

### Test Code
```
const { expect } = require('chai');
const { starknet } = require('hardhat');
const { stark: { randomAddress } } = require('starknet');

describe('Proxy contract', function () {
  this.timeout(600_000);
  let account;
  let accountAddress;
  let contract;

  before(async function () {
    account = await starknet.deployAccount('OpenZeppelin');
    accountAddress = BigInt(account.address);
  });

  beforeEach(async function () {
    console.info('deploying proxy contract...');
    const contractFactory = await starknet.getContractFactory('Proxy');
    contract = await contractFactory.deploy({ proxy_admin: account.address });
  });

  describe('setClassHash', function () {
    it('should set the class hash', async function () {
      const newClassHash = randomAddress();
      await account.invoke(contract, 'setClassHash', { class_hash: newClassHash });

      const { class_hash } = await contract.call('getClassHash');
      expect(BigInt(class_hash).toString()).to.eql(BigInt(newClassHash).toString());
    });

    it('should fail if not called by the admin', async function () {
      const newClassHash = randomAddress();
      try {
        await contract.invoke('setClassHash', { class_hash: newClassHash });
        expect.fail('setClassHash by non admin should fail');
      } catch (error) {
        expect(error.message).to.deep.contain('Proxy: caller is not admin');
      }
    });
  });
});

```

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

- Closes #131:
* `starknetChainId` is set inside `getNetwork()` found in `src/util.ts`.
* Fixes the error `Error: AssertionError: Unsupported chain ID: undefined.`  found when running,
```sh
npx hardhat starknet-invoke --contract contract --function increase_balance --address $ADDRESS --inputs "10 20" --wallet MyWallet --starknet-network myNetwork
 ```
## Development related changes

- NA
## Checklist:

-   [x] Formatted the code
-   [x] No linter errors
-   [x] Tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [X] Rebased to the last commit of the target branch (or merged it into my branch)
-   [ ] Documented the changes
-   [x] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.sh`)
-   [x] Linked issues which this PR resolves
-   [x] Created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example):https://github.com/Shard-Labs/starknet-hardhat-example/pull/72
    -   [x] Modified `test.sh` to use my example repo branch
    -   [x] Restored `test.sh` to to use the original branch (after the example repo PR has been merged)
-   [x] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

**Review by FabijanC — COMMENTED** [review]
Does it work with both alpha-goerli and devnet?

**Review by FabijanC — COMMENTED** [review]
Actually thinking about it, better question would be if it works with mainnet, since this looks like it will always default to `TESTNET` if using `hardhat run`. Is there another way of supporting this?

**Review by FabijanC — APPROVED** [review]

**Review by FabijanC — CHANGES_REQUESTED** [review]
As agreed elsewhere, expand testing of both wallet-test and hardhat-run (include relevant example repo changes in a PR to that repo)

File: test/general-tests/hardhat-run/check.sh [hunk 1]
```diff
@@ -4,3 +4,4 @@ set -e
 
 npx hardhat starknet-compile contracts/contract.cairo
 npx hardhat run --no-compile scripts/deploy.ts
+npx hardhat run --no-compile scripts/quick-script.ts
```
Comment (FabijanC) [inline test/general-tests/hardhat-run/check.sh:7]: Now that we have this quick-script.ts, I think there's no need to be calling deploy.ts as well

File: scripts/test.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ CONFIG_FILE_NAME="hardhat.config.ts"
 
 # setup example repo
 rm -rf starknet-hardhat-example
-git clone -b plugin --single-branch git@github.com:Shard-Labs/starknet-hardhat-example.git
+git clone -b bug-fix-invalid-signature --single-branch git@github.com:Shard-Labs/starknet-hardhat-example.git
```
Comment (FabijanC) [inline scripts/test.sh:12]: Now that your example repo PR has been merged, you can restore this.

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: test/general-tests/hardhat-run/check.sh (hunk 1 above)
Comment (Nathan-SL) [reply test/general-tests/hardhat-run/check.sh:7]: Yeah, makes sense.

**Pull request merged**